### PR TITLE
`bayesAB` function optionally returns samples based on KDE.

### PR DESCRIPTION
**Pull request by ghost**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
`bayesAB` function optionally returns samples based on KDE(Kernel Density Estimation).
...

Detailed description / Documentation draft:
`bayesAB` function calculates winning probabilities for each variants.

In addition to the winning probabilities, 

I added a feature returning samples based on KDE for the visualization like below graphs.
image

If you want to get the samples to draw graphs, you can add the last optional parameter.
`bayesAB(..., 1)`

If the last optional parameter is on, `bayesAB` returns the sample values.
```
{
  "data": [
    {
      "variant_name": "Control",
      "x": 10000,
      "y": 1000,
      "beats_control": 0,
      "to_be_best": 0,
      "samples": {              // The sample values
        "x": [
          0.08636688892346736, ... , 0.11443696706784429
        ],
        "y": [
          0.00015277701881408394, ... , 0.00004490102724527739
        ]
      }
    },
    ...
  ]
}
```
...

### CODE REVIEW

File: src/Functions/density.cpp [hunk 1]
```diff
@@ -0,0 +1,235 @@
+/*
+ *  This file is copied from R core to avoid including whole R core.
+ *
+ *  Copyright (C) 1997--2019   The R Core Team
+ *  Copyright (C) 1995, 1996   Robert Gentleman and Ross Ihaka
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
```
Comment (akuzm) [inline src/Functions/density.cpp:8]: We cannot accept the code under GPL licenses. MIT, BSD, Apache should be OK, if I recall correctly. You'll have to find another implementation, or write this code yourself.

File: .gitmodules [hunk 2]
```diff
@@ -180,3 +180,6 @@
 [submodule "contrib/stats"]
 	path = contrib/stats
 	url = https://github.com/kthohr/stats.git
+[submodule "contrib/eigen"]
+	path = contrib/eigen
+	url = https://gitlab.com/libeigen/eigen.git
```
Comment (akuzm) [inline .gitmodules:185]: eigen has some GPL parts as well, we have to define EIGEN_MPL2_ONLY

https://gitlab.com/libeigen/eigen/-/blob/master/COPYING.README

**Comment by achimbab** [comment]
@akuzm 
I will fix these codes as soon as possible. Thank you.

**Comment by nikitamikhaylov** [comment]
@achimbab I will convert this PR to draft, because it won't be merged soon.

**Comment by achimbab** [comment]
@nikitamikhaylov 
Hello, I will close this PR. Sorry.
I'll create another PR when I can work on it.
Thank you.

**Pull request closed without merging**